### PR TITLE
API v3.1 additions, changes base URLs, adds tests, bumps version

### PR DESCRIPTION
**Pull request by superdevx64**
* adds 'Download check' endpoint + test
* bumps base URL versions to 3.1
* bumps version of library to 1.2.0

Ciaran edit:
* Major version bump
* Mostly linting changes - worth going through by commit!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## v1.2.0, <DATE_TO_GO_HERE>
```
Comment (superdevx64) [inline CHANGELOG.md:1]: Date will go here when release confirmed

**Review by ciaran16 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## v2.0.0, <DATE_TO_GO_HERE>
```
Comment (nickpellant) [inline CHANGELOG.md:1]: Date still to be inserted.

File: lib/onfido/configuration.rb [hunk 3]
```diff
@@ -36,12 +36,10 @@ def logger
     end
 
     def endpoint
-      region_host = region ? REGION_HOSTS[region.downcase.to_sym] : "api.onfido.com"
-      unless region_host
-        raise "The region \"#{region.downcase}\" is not currently supported"
-      end
+      region_host = region ? REGION_HOSTS[region.downcase.to_sym] : 'api.onfido.com'
```
Comment (nickpellant) [inline lib/onfido/configuration.rb:39]: As this defaults to v3.1 and `api.onfido.com` doesn't fully support that version yet, will this cause issues as our fallback?

**Review by nickpellant — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (ciaran16) [reply CHANGELOG.md:1]: I'll update this in my MR

File: lib/onfido/configuration.rb (hunk 3 above)
Comment (ciaran16) [reply lib/onfido/configuration.rb:39]: Fixed in my MR (which merges into this one)

**Review by Phoebe-B — APPROVED** [review]

**Pull request merged**